spring-integration-core/src/main/java/org/springframework/integration/context/CustomConversionServiceFactoryBean.java

```diff
@@ -30,11 +32,22 @@
  */
 class CustomConversionServiceFactoryBean extends ConversionServiceFactoryBean {
 
+	public static final String MONGO_BASIC_QUERY_CONVERTER_CLASS_NAME = "org.springframework.integration.mongodb.support.StringToBasicQueryConverter";
+	private static final boolean isMongodbIntegrationPresent = ClassUtils.isPresent(MONGO_BASIC_QUERY_CONVERTER_CLASS_NAME, CustomConversionServiceFactoryBean.class.getClassLoader());
+
 	@Override
 	public ConversionService getObject() {
 		ConversionService service = super.getObject();
 		if (service instanceof GenericConversionService) {
 			((GenericConversionService) service).removeConvertible(Object.class, Object.class);
+			if (isMongodbIntegrationPresent) {
+				try {
+					Class<?> stringToBasicQueryConverterCls = Class.forName(MONGO_BASIC_QUERY_CONVERTER_CLASS_NAME);
+					((GenericConversionService) service).addConverter((GenericConverter) stringToBasicQueryConverterCls.newInstance());
```

Comment:
No, that's not a solution for us. This is some kind of package tangle just for particular case, when we can just evaluate expression to the `Object` and check its type to `String` or `BasicQuery`.

This customization just for the `BasicQuery` use-case in the common `ConversionService` doesn't provide much value and even gives some extra noise. That is much simpler to create `BasicQuery` manually from String SpEL result.